### ISSUE TITLE
Issues with proj >= 6

### ISSUE THREAD

**Issue by philippjfr** [SciTools/cartopy#1441]
### Description

When using cartopy with `proj>=6` (which is currently the default on conda-forge) a number of issues arise. Until these can be resolved it would be great if conda packages could pin to an earlier version of `proj`. The main issues can be summarized as:

* Vector projecting does not work correctly
* `RotatedPole` projections do not work correctly

<img width="256" alt="Screen Shot 2020-01-13 at 7 19 50 PM" src="https://user-images.githubusercontent.com/1550771/72281047-f6362200-3639-11ea-997e-e0ab6331c253.png">
<img width="591" alt="Screen Shot 2020-01-13 at 7 19 56 PM" src="https://user-images.githubusercontent.com/1550771/72281048-f6362200-3639-11ea-8c65-80dd70efe39e.png">
<img width="323" alt="Screen Shot 2020-01-13 at 7 20 10 PM" src="https://user-images.githubusercontent.com/1550771/72281049-f6ceb880-3639-11ea-9b82-23a744079442.png">

### proj==6.2.1

<img width="312" alt="Screen Shot 2020-01-13 at 7 21 37 PM" src="https://user-images.githubusercontent.com/1550771/72281050-f6ceb880-3639-11ea-8cdf-02dfeeb0349d.png">

#### Code to reproduce

[Here is a notebook](https://anaconda.org/philippjfr/cartopy_issues/notebook) which reproduces all the issues I've found (there may well be more issues).

<details>
 <summary>conda create -n cartopy_test -c conda-forge cartopy panel notebook</summary>

### Operating system

OSX Mojave Version 10.14.4

### Cartopy version

Cartopy 0.17.0

### conda list

```
appnope                   0.1.0                 py37_1000    conda-forge
attrs                     19.3.0                     py_0    conda-forge
backcall                  0.1.0                      py_0    conda-forge
bleach                    3.1.0                      py_0    conda-forge
bokeh                     1.4.0                    pypi_0    pypi
ca-certificates           2019.11.28           hecc5488_0    conda-forge
cartopy                   0.17.0          py37h4076dca_1009    conda-forge
certifi                   2019.11.28               py37_0    conda-forge
cffi                      1.13.2           py37h33e799b_0    conda-forge
chardet                   3.0.4                 py37_1003    conda-forge
cryptography              2.8              py37hafa8578_1    conda-forge
cycler                    0.10.0                     py_2    conda-forge
decorator                 4.4.1                      py_0    conda-forge
defusedxml                0.6.0                      py_0    conda-forge
entrypoints               0.3                   py37_1000    conda-forge
freetype                  2.10.0               h24853df_1    conda-forge
geos                      3.8.0                h4a8c4bd_0    conda-forge
idna                      2.8                   py37_1000    conda-forge
importlib_metadata        1.4.0                    py37_0    conda-forge
ipykernel                 5.1.3            py37h5ca1d4c_0    conda-forge
ipython                   7.11.1           py37h5ca1d4c_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jedi                      0.15.2                   py37_0    conda-forge
jinja2                    2.10.3                     py_0    conda-forge
jpeg                      9c                h1de35cc_1001    conda-forge
jsonschema                3.2.0                    py37_0    conda-forge
jupyter_client            5.3.4                    py37_0    conda-forge
jupyter_core              4.6.1                    py37_0    conda-forge
kiwisolver                1.1.0            py37ha1b3eb9_0    conda-forge
libblas                   3.8.0               14_openblas    conda-forge
libcblas                  3.8.0               14_openblas    conda-forge
libcxx                    9.0.1                         1    conda-forge
libffi                    3.2.1             h6de7cb9_1006    conda-forge
libgfortran               4.0.0                         2    conda-forge
liblapack                 3.8.0               14_openblas    conda-forge
libopenblas               0.3.7                h3d69b6c_6    conda-forge
libpng                    1.6.37               h2573ce8_0    conda-forge
libsodium                 1.0.17               h01d97ff_0    conda-forge
libtiff                   4.1.0                ha78913b_3    conda-forge
llvm-openmp               9.0.1                h28b9765_1    conda-forge
lz4-c                     1.8.3             h6de7cb9_1001    conda-forge
markdown                  3.1.1                    pypi_0    pypi
markupsafe                1.1.1            py37h0b31af3_0    conda-forge
matplotlib-base           3.1.2            py37h11da6c2_1    conda-forge
mistune                   0.8.4           py37h0b31af3_1000    conda-forge
more-itertools            8.1.0                      py_0    conda-forge
nbconvert                 5.6.1                    py37_0    conda-forge
nbformat                  5.0.3                      py_0    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
notebook                  6.0.1                    py37_0    conda-forge
numpy                     1.17.3           py37hde6bac1_0    conda-forge
olefile                   0.46                       py_0    conda-forge
openssl                   1.1.1d               h0b31af3_0    conda-forge
owslib                    0.19.0                     py_2    conda-forge
packaging                 20.0                     pypi_0    pypi
pandoc                    2.9.1.1                       0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
panel                     0.7.0                    pypi_0    pypi
param                     1.9.2                    pypi_0    pypi
parso                     0.5.2                      py_0    conda-forge
pexpect                   4.7.0                    py37_0    conda-forge
pickleshare               0.7.5                 py37_1000    conda-forge
pillow                    7.0.0            py37h918e99a_0    conda-forge
pip                       19.3.1                   py37_0    conda-forge
proj                      6.2.1                h773a61f_0    conda-forge
prometheus_client         0.7.1                      py_0    conda-forge
prompt_toolkit            3.0.2                      py_0    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
pycparser                 2.19                     py37_1    conda-forge
pyct                      0.4.6                    pypi_0    pypi
pyepsg                    0.4.0                      py_0    conda-forge
pygments                  2.5.2                      py_0    conda-forge
pykdtree                  1.3.1           py37h3b54f70_1002    conda-forge
pyopenssl                 19.1.0                   py37_0    conda-forge
pyparsing                 2.4.6                      py_0    conda-forge
pyproj                    2.4.2.post1      py37h03a428a_0    conda-forge
pyrsistent                0.15.7           py37h0b31af3_0    conda-forge
pyshp                     2.1.0                      py_0    conda-forge
pysocks                   1.7.1                    py37_0    conda-forge
python                    3.7.6                h5c2c468_2    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
pytz                      2019.3                     py_0    conda-forge
pyviz-comms               0.7.2                    pypi_0    pypi
pyyaml                    5.3                      pypi_0    pypi
pyzmq                     18.1.1           py37h4bf09a9_0    conda-forge
readline                  8.0                  hcfe32e1_0    conda-forge
requests                  2.22.0                   py37_1    conda-forge
scipy                     1.4.1            py37h82752d6_0    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                45.0.0                   py37_1    conda-forge
shapely                   1.6.4           py37h999ffa5_1007    conda-forge
six                       1.13.0                   py37_0    conda-forge
sqlite                    3.30.1               h93121df_0    conda-forge
terminado                 0.8.3                    py37_0    conda-forge
testpath                  0.4.4                      py_0    conda-forge
tk                        8.6.10               hbbe82c9_0    conda-forge
tornado                   6.0.3            py37h0b31af3_0    conda-forge
traitlets                 4.3.3                    py37_0    conda-forge
urllib3                   1.25.7                   py37_0    conda-forge
wcwidth                   0.1.8                      py_0    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.33.6                   py37_0    conda-forge
xz                        5.2.4             h1de35cc_1001    conda-forge
zeromq                    4.3.2                h6de7cb9_2    conda-forge
zipp                      0.6.0                      py_0    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge
zstd                      1.4.4                he7fca8b_1    conda-forge
```

</details>

**Comment by jbednar** [comment]
That looks highly alarming!

**Comment by ocefpaf** [comment]
It may be related to `pyproj` and not necessarily `proj >6` this combination works:

```
proj4                     6.1.0                he751ad9_2    conda-forge
pyproj                    2.2.0            py37hc44880f_0    conda-forge
```

See https://nbviewer.jupyter.org/gist/ocefpaf/6766437629eb885bf2bdd9565696d7f1

I'll create a package with `pyproj 2.2.0` and latest `proj` just to be sure.

Pinging @snowman2 here for awareness.

**Comment by snowman2** [comment]
`pyproj` is not a dependency of `cartopy`, so must be a cartopy PROJ interaction issue.

**Comment by ocefpaf** [comment]
> `pyproj` is not a dependency of `cartopy`, so must be a cartopy PROJ interaction issue.

Yes. I just noticed that after I sent the issue. Sorry for the false alarm.

**Comment by greglucas** [comment]
Looks like the example notebook works as expected with the current master Cartopy, so one option if using conda-forge is to `pip install git+https://github.com/SciTools/cartopy.git` to get Cartopy working with the currently pinned PROJ6 version.

It looks like the [conda feedstock](https://github.com/conda-forge/cartopy-feedstock) was updated in November to enable PROJ6 to be used (but not updating the Cartopy release branch to include the proj6 updated code which isn't yet in a numbered release). I'm not sure if that can be reverted to enable a lower PROJ pin again...

**Comment by philippjfr** [comment]
Bumping the build number and adding the pin would certainly be good. How far off is a new cartopy release?

**Comment by greglucas** [comment]
I think @dopplershift or @QuLogic are currently the ones to comment on how far off Cartopy is from a new release version.

@ocefpaf it looks like you bumped the previous feedstock version, are you able to change the pin on PROJ and bump the release?

**Comment by ocefpaf** [comment]
> @ocefpaf it looks like you bumped the previous feedstock version, are you able to change the pin on PROJ and bump the release?

Sorry. I'm be away for a few days and I won't be able to fix this.

Also, ideally we should backport the changes so we can keep using a modern proj. Or conflicts may arise while installing cartopy with other packages.

**Comment by Sieboldianus** [comment]
I am also seeing issues with cartopy installed from conda-forge (0.17.0, py37h4331854_1007) and Proj4 (6.1.1, hc80f0dc_1)

```python
cartopy.crs.epsg(32636)
```

Output:
```
Traceback (most recent call last):

  File "...miniconda3/envs/jupyter_env/lib/python3.7/xml/etree/ElementTree.py", line 1630, in feed
    self.parser.Parse(data, 0)

ExpatError: mismatched tag: line 33, column 4


During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "...miniconda3/envs/jupyter_env/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3319, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "<ipython-input-3-5f6c8f03c3a8>", line 1, in <module>
    cartopy.crs.epsg(32636)

  File "...miniconda3/envs/jupyter_env/lib/python3.7/site-packages/cartopy/crs.py", line 2540, in epsg
    return cartopy._epsg._EPSGProjection(code)

  File "...miniconda3/envs/jupyter_env/lib/python3.7/site-packages/cartopy/_epsg.py", line 69, in __init__
    x0, x1, y0, y1 = projection.domain_of_validity()

  File "...miniconda3/envs/jupyter_env/lib/python3.7/site-packages/pyepsg.py", line 205, in domain_of_validity
    gml = ET.fromstring(xml)

  File "...miniconda3/envs/jupyter_env/lib/python3.7/xml/etree/ElementTree.py", line 1315, in XML
    parser.feed(text)

  File "...miniconda3/envs/jupyter_env/lib/python3.7/xml/etree/ElementTree.py", line 1632, in feed
    self._raiseerror(v)

  File "...miniconda3/envs/jupyter_env/lib/python3.7/xml/etree/ElementTree.py", line 1531, in _raiseerror
    raise err

  File "<string>", line unknown
ParseError: mismatched tag: line 33, column 4
```

I have the feeling this is related to this issue, if not: apologies! I'll open a new issue then..

**Comment by greglucas** [comment]
That error is actually related to issue #1437, which is currently waiting on a fix in the epsg.io web server.

**Comment by QuLogic** [comment]
These bugs are fixed in `master`, and I tagged `v0.18.0b1` as a beta release. You can follow the progress to get out the final release in #1458.
